akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -148,6 +148,16 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
       event.getClass.getName, seqNr, persistenceId, cause.getMessage)
   }
 
+  /**
+    * Called when the size of the internal stash is exceeds the capacity of the Stash
+    * 
+    * @param message the message which failed to enqueue
+    */
+  protected def onStashOverflow(message: Any): Unit = {
```

Comment:
we must protect things while invoking this method in case user code does something bad, e.g. `persist` should not be allowed